src/backend/tests/it/utils/assertion.rs

```diff
@@ -19,6 +20,25 @@ pub fn assert_custom_tokens_eq(
     assert_eq!(results_tokens.len(), expected_tokens.len());
 
     for (token, expected) in results_tokens.iter().zip(expected_tokens.iter()) {
-        assert!(expected == token);
+        assert!(expected.token == token.token);
+        assert_eq!(expected.enabled, token.enabled);
+    }
+}
+
+pub fn assert_some_tokens_timestamp<T>(tokens: Vec<T>)
+where
+    T: TokenTimestamp,
+{
+    for token in tokens.iter() {
+        assert!(token.get_timestamp().is_some());
```

Comment:
```suggestion
        assert!(token.get_timestamp().is_some(), "Token has no timestamp: {}", token);
```
This will print which token has failed the assertion.  You will need to add the Debug trait to the type but this is generally a useful trait!